packages/test-base/src/androidTest/kotlin/io/realm/kotlin/test/shared/RealmInMemoryTests.kt

```diff
@@ -0,0 +1,243 @@
+package io.realm.kotlin.test.shared
+
+import io.realm.kotlin.Configuration
+import io.realm.kotlin.Realm
+import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.entities.Sample
+import io.realm.kotlin.ext.query
+import io.realm.kotlin.internal.RealmImpl
+import io.realm.kotlin.internal.platform.runBlocking
+import io.realm.kotlin.test.platform.PlatformUtils
+import io.realm.kotlin.test.util.TestHelper
+import junit.framework.AssertionFailedError
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import kotlin.random.Random
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.fail
+
+class RealmInMemoryTests {
```

Comment:
I know we didn't check for this in Java, but having a test checking that files are actually deleted on disk would be helpful.